File: android/app/src/main/java/com/cuhk/travelligent/CommentFragment.java
package com.cuhk.travelligent;

import android.content.Context;
import android.os.Bundle;
import android.os.Parcelable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.io.Serializable;
import java.util.List;

import io.swagger.client.models.CommentDto;

class CommentFragmentSerializableArg implements Serializable {

    public List<CommentDto> comments;

    public CommentFragmentSerializableArg(List<CommentDto> comments) {
        this.comments = comments;
    }

}

/**
 * A fragment representing a list of Items.
 */
public class CommentFragment extends Fragment {

    // TODO: Customize parameter argument names
    private static final String ARG_COLUMN_COUNT = "column-count";
    private static final String ARG_ARG = "arg";

    // TODO: Customize parameters
    private int mColumnCount = 1;
    private List<CommentDto> comments;

    /**
     * Mandatory empty constructor for the fragment manager to instantiate the
     * fragment (e.g. upon screen orientation changes).
     */
    public CommentFragment() {
    }

    // TODO: Customize parameter initialization
    @SuppressWarnings("unused")
    public static CommentFragment newInstance(int columnCount, List<CommentDto> comments) {
        CommentFragment fragment = new CommentFragment();
        Bundle args = new Bundle();
        args.putInt(ARG_COLUMN_COUNT, columnCount);
        args.putSerializable(ARG_ARG, new CommentFragmentSerializableArg(comments));
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (getArguments() != null) {
            mColumnCount = getArguments().getInt(ARG_COLUMN_COUNT);
            comments = ((CommentFragmentSerializableArg) getArguments().getSerializable(ARG_ARG)).comments;

            System.out.println(comments);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_comment_list, container, false);

        // Set the adapter
        if (view instanceof RecyclerView) {
            Context context = view.getContext();
            RecyclerView recyclerView = (RecyclerView) view;
            if (mColumnCount <= 1) {
                recyclerView.setLayoutManager(new LinearLayoutManager(context));
            } else {
                recyclerView.setLayoutManager(new GridLayoutManager(context, mColumnCount));
            }
            recyclerView.setAdapter(new MyCommentRecyclerViewAdapter(comments));
        }
        return view;
    }

}

File: android/app/src/main/java/com/cuhk/travelligent/component/CommentFragment.kt
package com.cuhk.travelligent.component

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.cuhk.travelligent.R
import kotlinx.android.synthetic.main.fragment_comment.view.*

// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
private const val ARG_CREATE_USER_NAME = "param1"
private const val ARG_CONTENT = "param2"

/**
 * A simple [Fragment] subclass.
 * Use the [CommentFragment.newInstance] factory method to
 * create an instance of this fragment.
 *
 */
class CommentFragment : Fragment() {

    private var createUserName: String? = null
    private var content: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        arguments?.let {
            createUserName = it.getString(ARG_CREATE_USER_NAME)
            content = it.getString(ARG_CONTENT)
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_comment, container, false)

        view.create_user_name.text = createUserName
        view.content.text = content

        return view
    }


    companion object {
        /**
         * Use this factory method to create a new instance of
         * this fragment using the provided parameters.
         *
         * @param createUserName Parameter 1.
         * @param content Parameter 2.
         * @return A new instance of fragment CommentFragment.
         */
        @JvmStatic
        fun newInstance(createUserName: String, content: String) =
            CommentFragment().apply {
                arguments = Bundle().apply {
                    putString(ARG_CREATE_USER_NAME, createUserName)
                    putString(ARG_CONTENT, content)
                }
            }
    }
}
